File: CSharp/OpenTraceability/Interfaces/IMasterDataMapper.cs
using System;
namespace OpenTraceability.Interfaces
{
	public interface IMasterDataMapper
	{
		string Map(IVocabularyElement vocab);
        IVocabularyElement Map(Type T, string value);
		IVocabularyElement Map<T>(string value) where T : IVocabularyElement;
	}
}

File: Java/OpenTraceability/src/main/kotlin/opentraceability/interfaces/imasterdatamapper.kt
package opentraceability.interfaces

import java.lang.reflect.Type
import kotlin.reflect.KClass

interface IMasterDataMapper {
    fun map(vocab: IVocabularyElement): String
    fun map(t: KClass<*>, value: String): IVocabularyElement
    fun <T: IVocabularyElement> map(value: String): IVocabularyElement
}
